doc/NAC/PnacSonicHLD.md

```diff
@@ -0,0 +1,371 @@
+* [2 Functionality](#2-Functionality)
+   * [2.1 Target Deployment Use Cases](#21-Target-Deployment-Use-Cases)
+* [3 Design](#3-Design)
+  * [3.1 Options to Block or allows port traffic based on Authentication status of the port](#31-Options-to-Block-or-allows-port-traffic-based-on-Authentication-status-of-the-port)
+     * [3.1.1 Options-1 Allowing or Blocking user traffic by using RSTP STP states](#311-Options-1-Allowing-or-Blocking-user-traffic-by-using-RSTP-STP-states)
+     * [3.1.2 Option-2 Allowing or Blocking user traffic by configuring L2-ACL filters](#312-Option-2-Allowing-or-Blocking-user-traffic-by-configuring-L2-ACL-filters)
+     * [3.1.3 Option-3 Allowing or Blocking user traffic by configuring learn mode](#313-Option-3-Allowing-or-Blocking-user-traffic-by-configuring-learn-mode) 
+   * [3.2 PNAC software architecture](#32-PNAC-software-architecture)
+      * [3.2.1 Hostapd initialisation steps](#321-Hostapd-initialisation-steps)
+      * [3.2.2 Actions performed by hostapd wrapper](#322-Actions-performed-by-hostapd-wrapper)
+      * [3.2.3 Actions Triggered Based on CLI configuration changes](#323-Actions-Triggered-Based-on-CLI-configuration-changes)
+      * [3.2.4 Actions performed after Receiving state change notification from hostapd](#324-Actions-performed-after-receiving-state-change-notification-from-hostapd)
+      * [3.2.5 Actions performed when interface operational status changes to DOWN](#325-Actions-performed-when-interface-operational-status-changes-to-DOWN)
+      * [3.2.6 Actions performed when interface operational state changes to UP](#326-Actions-performed-when-interface-operational-state-changes-to-UP)
+      * [3.2.7 Sequence of Events flowchart](#327-Sequence-of-Events-flowchart)
+   * [3.3 DB Changes](#33-DB-Changes)
+     * [3.3.1 ConfigDB Schema Changes](#331-ConfigDB-Schema-Changes)
+     * [3.3.2 App DB Changes](#332-App-DB-Changes)
+  * [3.4 CLI Commands](#34-CLI-Commands)
+* [4 Serviceability and Debug](#4-Serviceability-and-Debug)
+* [5 Warm Reboot](#5-Warm-Reboot)
+* [6 Scalability](#6-Scalability)
+* [7 Unit Test](#7-Unit-test)
+* [8 To be done in future release](#8-To-be-done-in-future-release)
+# Revision
+| Rev |     Date    |       Author       | Change Description                  |
+|:---:|:-----------:|:-------------------|:-----------------------------------|
+| 0.1 |  10/03/2023 | Kishor Kulkarni, <br> Keerthi Kumar Thovi| Initial version                     |
+
+# About this Document
+This document describes the design details of the Port Based Network Access Control (PNAC) feature.
+PNAC enables switches to control port access only for Authenticated users while denying access for unauthenticated users. The core to this feature is IEEE protocol 802.1x protocol. Port-based network access control allows a network administrator to restrict the use of LAN service access port only for authenticated and authorised users those are trying to get access into the network. 
```

Comment:
Can you please specify which version of the standard being implemented; it will help review the HLD.

Reply:
IEEE Std 802.1X-2020 version is used as a reference for implementation ,

Reply:
Which version of the opensource hostapd are you using ? Does it have IEEE Std 802.1X-2020 (State Machines etc) implementation ?

Reply:
There is an implementation of  IEEE 802.1X, which is part of community SONiC MACSEC docker. Is this design going to leverage the same ?

Reply:
Yes we are using hostapd and version is 2.9.0.  
PNAC design is not reusing SONIC MACSEC decker.

Reply:
No, design will have dedicated docker container for PNAC feature, since we don't want to create dependency between features.